src/Microsoft.AspNet.Mvc.DataAnnotations/DataAnnotationsModelValidator.cs

```diff
@@ -38,6 +148,8 @@ public IEnumerable<ModelValidationResult> Validate(ModelValidationContext valida
                 MemberName = memberName
             };
 
+            var metadataProvider = (IModelMetadataProvider)context.GetService(typeof(IModelMetadataProvider));
```

Comment:
This works? Do we do this anywhere else? If not, we shouldn't be using the data annotation's validation context to locate an MVC service.